src/libAtomVM/nifs.c

```diff
@@ -3553,6 +3560,56 @@ static term nif_atomvm_read_priv(Context *ctx, int argc, term argv[])
     return result;
 }
 
+static term nif_atomvm_set_time_of_day(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc);
+
+    VALIDATE_VALUE(argv[0], term_is_atom);
+    VALIDATE_VALUE(argv[1], term_is_any_integer);
+
+#ifdef HAVE_SETIMEOFDAY
+
+    term unit = argv[0];
+    avm_int64_t value = term_maybe_unbox_int64(argv[1]);
+
+    struct timeval tp;
+    if (unit == SECOND_ATOM) {
+        tp.tv_sec = value;
+        tp.tv_usec = 0L;
+
+    } else if (unit == MILLISECOND_ATOM) {
+        tp.tv_sec = value / 1000;
+        tp.tv_usec = (value % 1000) * 1000;
+
+    } else if (unit == MICROSECOND_ATOM) {
+        tp.tv_sec = value / 1000000;
+        tp.tv_usec = value % 1000000;
+
+    } else {
+        RAISE_ERROR(BADARG_ATOM);
+    }
+
+    struct timezone tz = {
+        .tz_minuteswest = 0,
+        .tz_dsttime = 0
+    };
+    int res = settimeofday(&tp, &tz);
+    if (res != 0) {
+        if (UNLIKELY(memory_ensure_free(ctx, TUPLE_SIZE(2)) != MEMORY_GC_OK)) {
+            RAISE_ERROR(OUT_OF_MEMORY_ATOM);
+        }
+        term error = term_alloc_tuple(2, &ctx->heap);
+        term_put_tuple_element(error, 0, ERROR_ATOM);
+        term_put_tuple_element(error, 1, posix_errno_to_term(errno, ctx->global));
+        RAISE_ERROR(error);
+    } else {
+        return OK_ATOM;
+    }
+#else
+    RAISE_ERROR(UNIMPLEMENTED_ATOM);
```

Comment:
There already is an error for unimplemented functions: `undef`

Approach with POSIX file functions is to simply not define the nif. The code will fail with `undef`.

Reply:
Fixed